### DISCUSSION TITLE
Eror saat prediksi data test

### DISCUSSION THREAD

**Discussion by AmriDomas** [teamalgoritma/community#739] [Q&A] [answered]
Dear All teaalgoritma,

Ada permaslahan saat prediksi data test di bawah ini

<img width="585" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/85354c82-cb8e-478e-afe2-3d6c62969930">

dengan random forest seperti di bawah ini

<img width="469" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/21129b23-3ad3-4d26-81a9-2a7303eea38b">

Apa yang harus saya perbaiki?

dan pada saat saya prediksi untuk data validasi pun bisa di lakukan

<img width="344" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/559c6bb8-e02e-4248-ad78-03810796e765">

Regards,
Muh Amri Sidiq

**Comment by amaliasf** [comment, marked as answer]
Dear @AmriDomas ,

Untuk kendala saat melakukan prediksi data test, padahal sudah berhasil melakukan prediksi data validasi mungkin bisa dipastikan terlebih dahulu apakah semua treatment atau tahap pre-processing pada data test sudah dilakukan sama persis sebagaimana data train.

Bisa diperhatikan struktur data pada `vaidation_scot` sudah sama persis dengan `scotty_test`. Khususnya bisa ditinjau pada kolom `month` (berdasarkan persan error yang muncul).

Silahkan boleh dicek dahulu Pak, kalau masih terkendala bisa diinfokan kembali. Semoga membantu dan menjawab pertanyaan Anda.

**Comment by AmriDomas** [comment]
Ok. terimakasih
